src/Uno.UI/UI/Xaml/Controls/MenuFlyout/MenuFlyoutPresenter.cs

```diff
@@ -1,13 +1,1008 @@
+			var nCount = (menuItems as ItemCollection).Size;
+			for (var i = 0; i < nCount; ++i)
+			{
+				MenuFlyoutItemBase item;
+				MenuFlyoutItem itemAsMenuItem;
+
+				item = Items[i] as MenuFlyoutItemBase;
+
+				itemAsMenuItem = item as MenuFlyoutItem;
+				if (itemAsMenuItem != null)
+				{
+					var desiredSize = (itemAsMenuItem as MenuFlyoutItem).GetKeyboardAcceleratorTextDesiredSize();
+					var desiredWidth = desiredSize.Width;
+					if (desiredWidth > maxItemKeyboardAcceleratorTextWidth)
+					{
+						maxItemKeyboardAcceleratorTextWidth = desiredWidth;
+					}
+				}
+			}
+
+			for (var i = 0; i<nCount; ++i)
+			{
+				MenuFlyoutItemBase item;
+				MenuFlyoutItem itemAsMenuItem;
+
+				item = Items[i] as MenuFlyoutItemBase;
+
+				itemAsMenuItem = item as MenuFlyoutItem;
+				if (itemAsMenuItem != null)
+				{
+					(itemAsMenuItem as MenuFlyoutItem).UpdateTemplateSettings(maxItemKeyboardAcceleratorTextWidth);
+				}
+			}
+		}
+
+		protected override void OnGotFocus(RoutedEventArgs pArgs)
+		{
+			var focusState = FocusState.Unfocused;
+
+			focusState = FocusState;
+
+			if (m_iFocusedIndex == -1 &&
+				focusState != FocusState.Unfocused)
+			{
+				// The MenuFlyoutPresenter gets focused for the first time right after it is opened.
+				// In this case we want to send focus to the first focusable item.
+				CycleFocus(true /* shouldCycleDown */, focusState);
+			}
+			else if (focusState == FocusState.Unfocused)
+			{
+				// A child element got focus, so make sure we keep m_iFocusedIndex in sync
+				// with it.
+				var focusedElement = FocusManager.GetFocusedElement() as DependencyObject;
+
+				// Since GotFocus is an async event, the focused element could be null if we got it
+				// after the popup closes, which clears focus.
+				if (focusedElement != null)
+				{
+					var focusedElementIndex = -1;
+					focusedElementIndex = IndexFromContainer(focusedElement);
+					if (focusedElementIndex != -1)
+					{
+						m_iFocusedIndex = focusedElementIndex;
+					}
+				}
+			}
+
+
+		}
+
+		void EnsureInitialFocusIndex()
+		{
+			if (m_iFocusedIndex == -1)
+			{
+				var focusedElement = FocusManager.GetFocusedElement();
+
+				if (this != focusedElement)
+				{
+					var menuItemsCount = 0;
+					IObservableVector<object> menuItems = Items;
+
+					menuItemsCount = Items.Count;
+
+					for (var i = 0; i < menuItemsCount; ++i)
+					{
+						var itemAsDependencyObject = Items[i] as DependencyObject;
+						MenuFlyoutItem menuItem;
+
+						menuItem = itemAsDependencyObject as MenuFlyoutItem;
+
+						if (menuItem == focusedElement)
+						{
+							m_iFocusedIndex = i;
+							break;
+						}
+					}
+
+					global::System.Diagnostics.Debug.Assert(m_iFocusedIndex != -1);
+				}
+			}
+		}
+
+		int GetPositionInSetHelper(MenuFlyoutItemBase item)
+		{
+			var returnValue = -1;
+
+			var presenter= item.GetParentMenuFlyoutPresenter();
+
+			if (presenter != null)
+			{
+				IObservableVector<object> items = presenter.Items;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the unused local variable 'items'.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5205492)